SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -0,0 +1,171 @@
+//==------- esimd_bit_ops.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu || level_zero
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for esimd default constructor.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <string>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+#include "../logger.hpp"
+#include "../type_coverage.hpp"
+
+using namespace type_coverage;
+using namespace logger;
+using namespace cl::sycl;
+using namespace sycl::ext::intel::experimental;
+using namespace sycl::ext::intel::experimental::esimd;
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataTypeT, int TestNumber, int NumElems>
+struct kernel_for_simd;
+
+const auto types =
```

Comment:
Why isn't this variable in the function scope of the `main`?

Reply:
I've moved it into separate geter-function in this commit: 4970bd2d73f69c409e7477cccff9a49e4b1da8f8